pulsar-common/src/main/java/org/apache/pulsar/common/api/proto/PulsarApi.java

```diff
@@ -1209,11 +1214,26 @@ public int getBatchIndex() {
       return batchIndex_;
     }
     
+    // repeated int64 ackSet = 5;
+    public static final int ACKSET_FIELD_NUMBER = 5;
+    private java.util.List<java.lang.Long> ackSet_;
+    public java.util.List<java.lang.Long>
+        getAckSetList() {
+      return ackSet_;
+    }
+    public int getAckSetCount() {
+      return ackSet_.size();
+    }
+    public long getAckSet(int index) {
+      return ackSet_.get(index);
+    }
+    
     private void initFields() {
       ledgerId_ = 0L;
       entryId_ = 0L;
       partition_ = -1;
       batchIndex_ = -1;
+      ackSet_ = java.util.Collections.emptyList();;
```

Comment:
Double `;;` You might want to do a global replacement since these are in multiple places.

Reply:
The PulsarApi.java is generated by a proto tool. So I won’t do it manually here, it's better to improve the generation tool.